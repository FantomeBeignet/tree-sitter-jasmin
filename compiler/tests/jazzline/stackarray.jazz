fn f(reg u64 x) -> reg u64
{
   inline int i;
   stack u64[3] t;
   reg u64 y;
   for i = 0 to 3 {
      t[i] = x;
   }

   for i = 0 to 3 {
      y += t[i];
   }

   return y;
}
