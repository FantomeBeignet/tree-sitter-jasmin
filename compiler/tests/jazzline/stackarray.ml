open Common

let () = parse_and_print "stackarray.jazz"  "f";
