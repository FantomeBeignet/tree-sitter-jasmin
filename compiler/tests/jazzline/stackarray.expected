proc f(y_160@uint64, x_135@uint64) =
{
 true && true
}
mov t_0_155@uint64 x_135@uint64;
mov t_1_156@uint64 x_135@uint64;
mov t_2_157@uint64 x_135@uint64;
adds NONE_____171@uint1 y_158@uint64 y_136@uint64 t_0_155@uint64;
adds NONE_____172@uint1 y_159@uint64 y_158@uint64 t_1_156@uint64;
adds NONE_____173@uint1 y_160@uint64 y_159@uint64 t_2_157@uint64;
{
 true && true
}
 